File: frontend/src/lib/components/definitions/rm-hacks.ts
import { ComponentDefinition } from '../types'
import { CommandBuilders } from '../utils'

export const rmHacksComponent: ComponentDefinition = {
  id: 'rm-hacks',
  name: 'rM Hacks',
  description: 'Split screen, custom pen sizes, and more',
  version: '0.0.11-pre3',
  author: 'asivery',

  dependencies: ['xovi', 'qt-resource-rebuilder'],

  checkInstalled: () =>
    CommandBuilders.chain(
      CommandBuilders.testExists('/home/root/xovi/exthome/qt-resource-rebuilder/zz_rmhacks.qmd'),
      CommandBuilders.testExists('/home/root/xovi/exthome/qt-resource-rebuilder/rmhacks', 'directory')
    ),

  install: () => {
    const baseUrl = 'https://github.com/asivery/rm-hacks-qmd/archive/refs/heads/master.zip'

    return [
      CommandBuilders.changeDirectory('/home/root'),
      {
        script: `./curl -Lo rm-hacks.zip ${baseUrl}`,
        description: 'Download rM Hacks',
      },
      {
        script: 'unzip -o -d rm-hacks rm-hacks.zip "rm-hacks-qmd-master/0.0.11-pre3/*"',
        description: 'Extract rM Hacks',
      },
      {
        script: 'cp -r rm-hacks/rm-hacks-qmd-master/0.0.11-pre3/* /home/root/xovi/exthome/qt-resource-rebuilder/',
        description: 'Install rM Hacks to xovi',
      },
      {
        script: 'rm -rf rm-hacks rm-hacks.zip',
        description: 'Clean up temporary files',
      },
    ]
  },

  uninstall: () => [
    CommandBuilders.removeFile('/home/root/xovi/exthome/qt-resource-rebuilder/zz_rmhacks.qmd'),
    CommandBuilders.removeDirectory('/home/root/xovi/exthome/qt-resource-rebuilder/rmhacks'),
  ],

  category: 'enhancement',
  tags: ['ui', 'productivity'],
  requiresReboot: false,
}

File: internal/component/definitions/rm_hacks.go
package definitions

import (
	"reManager/internal/commands"
	"reManager/internal/component"
)

var RmHacksComponent = &component.ComponentDefinition{
	ID:          "rm-hacks",
	Name:        "rM Hacks",
	Description: "Split screen, custom pen sizes, and more",
	Version:     "0.0.11-pre3",
	Author:      "asivery",

	Dependencies: []string{"xovi", "qt-resource-rebuilder"},

	CheckInstalled: func(ctx component.CommandContext) component.CommandResult {
		return commands.Chain(
			commands.TestExists("/home/root/xovi/exthome/qt-resource-rebuilder/zz_rmhacks.qmd", commands.PathFile),
			commands.TestExists("/home/root/xovi/exthome/qt-resource-rebuilder/rmhacks", commands.PathDirectory),
		)
	},

	Install: func(ctx component.CommandContext) []component.CommandResult {
		baseUrl := "https://github.com/asivery/rm-hacks-qmd/archive/refs/heads/master.zip"

		return []component.CommandResult{
			commands.ChangeDirectory("/home/root"),
			{
				Script:      "./curl -Lo rm-hacks.zip " + baseUrl,
				Description: "Download rM Hacks",
			},
			{
				Script:      `unzip -o -d rm-hacks rm-hacks.zip "rm-hacks-qmd-master/0.0.11-pre3/*"`,
				Description: "Extract rM Hacks",
			},
			{
				Script:      "cp -r rm-hacks/rm-hacks-qmd-master/0.0.11-pre3/* /home/root/xovi/exthome/qt-resource-rebuilder/",
				Description: "Install rM Hacks to xovi",
			},
			{
				Script:      "rm -rf rm-hacks rm-hacks.zip",
				Description: "Clean up temporary files",
			},
		}
	},

	Uninstall: func(ctx component.CommandContext) []component.CommandResult {
		return []component.CommandResult{
			commands.RemoveFile("/home/root/xovi/exthome/qt-resource-rebuilder/zz_rmhacks.qmd"),
			commands.RemoveDirectory("/home/root/xovi/exthome/qt-resource-rebuilder/rmhacks"),
		}
	},

	Category:       "enhancement",
	Tags:           []string{"ui", "productivity"},
	RequiresReboot: false,
}
